### PR TITLE
Alerting: Implement the Webex notifier

### LINKED ISSUE #11750: Add notifier for Webex Teams
This issue is to track support of notifications for Cisco [Webex Teams](https://www.webex.com/products/teams/) rooms (recently re-branded from Cisco Spark).

I sent a working PR for that in https://github.com/grafana/grafana/pull/11702, but Grafana currently does not accept new notifiers. This needs to be revisited once backend plugins are available.

### PR DESCRIPTION
**Pull request by gotjosh**
Closes https://github.com/grafana/grafana/issues/11750

Sadly, I tried to go the unification route where I wanted to use the upstream notifier as part of the messaging, but the images in notification made this impossible to work around.

That said, I do have a clear idea now of what we need to modify upstream to make this reality - for now, I'd like to ship this as is and use this exact notifier as the first guinea pig for the unification.

Screenshot 2022-11-08 at 20 08 06

TODO
- [x] Tests
- [x] Documentation
- [x] Alignment of settings with Prometheus Alertmanager

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/webex.go
Comment (gotjosh) [inline pkg/services/ngalert/notifier/channels/webex.go:39]: This is my most recent discovery - looking at the Webex documentation it seems that the recommended approach is to use a "Bot", the "Bot" comes with a Token that you can use to interact with the Webex rooms. 

As such, the required configuration appears to be: A bot token and which room you want to send alerts too (1 at a time).

I couldn't figure out a way to include the token/room as part of the URL as I previously thought, which means that we probably don't need to encrypt the URL and just the token?

File: pkg/services/ngalert/notifier/channels/webex_test.go
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/channels/webex_test.go:95]: does `111....` have any meaning? what does it test? probably use `strings.Repeat("1",4097)`?

Comment (gotjosh) [reply pkg/services/ngalert/notifier/channels/webex_test.go]: Generally, I'm not a fan of using code in my test assertions and would see and have explicitly what the result is, this is so minor and a function from the standard library (as opposed to something I wrote) so I'll fly with it.

It's fixed.

File: pkg/services/ngalert/notifier/channels/webex.go
Comment (grobinson-grafana) [inline pkg/services/ngalert/notifier/channels/webex.go:47]: I noticed here we return `settings` but on Line 62 we return `nil`?

File: pkg/services/ngalert/notifier/channels/webex.go
Comment (grobinson-grafana) [inline pkg/services/ngalert/notifier/channels/webex.go:109]: Should be `markdown` or `message`?

```suggestion
	Message string   `json:"message"`
```

File: pkg/services/ngalert/notifier/channels/webex.go
Comment (grobinson-grafana) [inline pkg/services/ngalert/notifier/channels/webex.go:34]: Link doesn't work for me

**Review by grobinson-grafana — APPROVED** [review]

File: pkg/services/ngalert/notifier/channels/webex.go
Comment (gotjosh) [reply pkg/services/ngalert/notifier/channels/webex.go:109]: It's weird, but it should markdown because that's the name that webex uses. They have 2 with are `message` and `markdown` and `markdown` is the markdown version of `message` 🤷 

I don't think we should expose this complexity to the user and instead allow to set the message (similar to other notifiers, that also call it "message' despite being something else in the payload)

File: pkg/services/ngalert/notifier/channels/webex.go
Comment (gotjosh) [reply pkg/services/ngalert/notifier/channels/webex.go:34]: It's not there yet, we need to wait for https://github.com/prometheus/alertmanager/pull/3132 to be merged, which I'll merge on Monday once I get another quick review from alex.

**Review by grobinson-grafana — APPROVED** [review]

**Review by grobinson-grafana — APPROVED** [review]

**Pull request merged**
